File: SE/assignment/module 03/5)string/Q12.cpp
//12.Write a program in C to find the number of times a given word 'is' appears in
//the given string.

#include<stdio.h>

main()
{
	char a[30];
	int i, count=0;
	
	printf("Enter the string:");  //taking input
	gets(a);
	
	for(i=0;a[i]!='\0';i++)
	
	{
		if (a[i]==' ' &&a[i+1]=='i' &&a[i+2]=='s' &&(a[i+3]==' ' ||a[i+3]=='\0'||a[i]=='.'))
		count++;
	}                                  //putting conditions
	printf("%d times 'is' appears in string ",count);
}

File: Python/Assignment/Module-2 (Fundamentals of python/Q12.py
#Write a Python program to count occurrences of a substring in a string.

n = input("Enter the string:")
s = input("Enter the substring:")

count=0
start=0
sl=0

for i in n:
    sl+=1

while start<sl:
    pos = n.find(s,start)
    if pos!=-1:
        count+=1
        start = pos+1
    else:
        break

print(f"overlapping occurence of '{s}'in '{n}':{count}")           

